GraphQL-TSC.md

```diff
@@ -162,24 +168,43 @@ When something cannot be decided by consensus or a formal vote is required, the
 
 The ballots may be public or private depending upon the situation, but the aggregate results should always be published.
 
-#### Minimum threshold for votes
+### Defining a quorum of "attending" members
+
+To balance preserving the voting ability of all TSC members with the desire for the voting process to remain efficient and avoid deadlocks, we define a valid quorum based on members' recent attendance of GraphQL Working Group meetings.
 
-Because we work in a distributed environment, the voting process must account for a range of time zones and schedules. A quorum of at least half (or supermajority when relevent) of the TSC voting members must cast a ballot in order for the vote to be valid. Once the voting threshold has been met, one of these two critera must be satisfied to conclude the vote:
+A quorum is a majority (more than half, or 2/3 for a supermajority vote) of the TSC *attending members*. A quorum must cast a ballot in order for a vote to be valid.
+
+A TSC *attending member* is a member who has attended one of the previous three meetings. Should a TSC member miss three consecutive meetings, they will no longer be counted when determining quorum (but may still vote). A member starts counting towards quorum as of attending a meeting.
+
+### Voting process
+
+Because we work in a distributed environment, the voting process must account for a range of time zones and schedules. Once the threshold of a quorum has been met and a vote is valid, one of these two critera must be satisfied to conclude a vote:
 
 * A notice is sent via email that the vote will conclude in three business days, reminding those who haven't voted that they should do so. The vote will conclude at the end of this time.
-* The election results will not change if additional TSC members vote.
+* The election results would not change with the addition of another vote.
 
-#### Ensuring efficient votes
+Once a valid vote is concluded, the result is determined by the number of votes recieved at that time (as opposed to the total number of TSC members):
 
-A TSC member must attend two of the previous five meetings to be considered a "TSC voting member" and be eligible to vote. Should a TSC member miss three or more of the prior five meetings, they will no longer vote or be counted toward the total number of voting members. This is to ensure broad participation as well as to avoid voting deadlocks.
+* For a simple majority (or supermajority), the votes in favor must exceed half (or 2/3) of the total number of votes.
+* For the Condorcet method, all votes received at the time the vote is concluded are considered.
 
-A member may regain their voting status by attending two meetings and may participate in votes in the second meeting.
+TSC members are not required to vote. An abstention vote is not considered part of the total set of votes, nor towards a voting quorum.
```

Comment:
I don’t feel this is quite right; this should be a little analogous to a “conflict of interest” situation - i.e. the quorum should be recalculated as if the abstaining member were not an attending member. This reads as if they don’t count towards reaching the quorum but without also (potentially) reducing the quorum.

We should also have a rule for the minimum size of a quorum; e.g. 2 people.

Reply:
I don't know that I follow your concern. My thought was that an abstain can't simply count as a vote that's not yes, since that would make it equivalent to a no. So it needs to not count as a vote at all such that "a majority of votes" still holds. Also, if someone abstains, that shouldn't trigger a vote quorum either. You need a quorum of yes/no votes to make a vote valid. 

Is there more to this that I've missed?


I think I like the idea of a minimum quorum, but wouldn't that only happen in the case of extreme absence, in which case it might be possible minimum quorum can't be reached and things deadlock?

Reply:
Since quorum is "more than half of attending" then for cases of very low attendance: 4 attending means needing 3 for quorum. 3 attending means 2 for quorum. 2 attending also means 2 for quorum. 1 attending means 1 for quorum.

Reply:
TL;DR: "nor towards a voting quorum" is ambiguous - does it mean that they would not count when calculating the quorum, or when calculating if the quorum was fulfilled?

Laying out a concrete example to explain my concern:

7 attending members.

Quorum is therefore 4.

4 attending members abstain.

Three remaining attending members vote (two for, one against).

No non-attending members vote.

You state abstentions are not considered "towards a voting quorum" so it's clear there's only 3 votes that count towards the quorum. However the quorum requirement is 4, therefore this vote is invalid due to too many abstentions? Effectively the abstention is a "no".

Therefore, I think that abstentions should be subtracted from the size of the pool used to calculate the quorum in the first place - 7 attending members less 4 abstentions is 3 remaining attending members. Quorum is thus 2, so the vote is valid, and the result (2 for, 1 against) is that the vote passes. This is similar to how someone would be removed from the quorum-determining pool in the case of conflict-of-interest.

Reply:
> wouldn't that only happen in the case of extreme absence, in which case it might be possible minimum quorum can't be reached and things deadlock?

Yes, I'd prefer a deadlock over a single person (out of 11!) getting the one and only vote on the subject. I think that at least two people should have to vote for a change for the change to occur - "favour no change," right? :wink:

The only exception to this would be voting on new TSC members - should we all attend a conference and get hit by a meteor (and thus unable to vote) then the sole remaining TSC member should be able to appoint a new committee such that votes on other subjects may continue.

Reply:
Hopefully my last edit navigates these concerns.

I added definitions for present/abstain/recused and their impact on quorum that way people can have an informed impact on the process. I think this avoids the concern with the size of quorum dropping to a very low number, even 1. "Abstain" (or no reply) does not allow a vote to occur since the threshold remains the same but hinders progress towards it, where present/recused would either make progress towards that threshold or lower the threshold accordingly - but those are intentionally provided replies, not implied.

The only scenario in which the threshold for quorum can reach 1 in an unintentional way is if all but one member have not attended 3 meetings. That effectively gives the remaining person full power, and I think this is a good thing. This is the hit-by-meteor scenario.